### ISSUE TITLE
Deduplicate SentryOptions default value tests

### ISSUE THREAD

**Issue by armcknight** [getsentry/sentry-cocoa#3564]
### Description

We have lots of test methods that test each option's default value in isolation: 
<img width="310" alt="image" src="https://github.com/getsentry/sentry-cocoa/assets/3241469/9daed284-305e-41b1-b724-aebde234dea1">

but we also have a big method that asserts all the default values at https://github.com/getsentry/sentry-cocoa/blob/42ef6baf82435c9e60f92974097657417ca4839a/Tests/SentryTests/SentryOptionsTest.m#L571-L647

Each of the individual methods instantiates a `SentryOptions` like so: `SentryOptions *options = [self getValidOptions:@{}];`, but we could just create one and call `assertDefaultValues` on it.

**Comment by philipphofmann** [comment]
I agree that this is suboptimal, but we rarely change these tests, and the options also hardly ever change. I would also be OK with keeping them as is, but fixing it also makes sense.
